microprofile/testing/testing/src/main/java/io/helidon/microprofile/testing/HelidonTestConfigSynthetic.java

```diff
@@ -173,6 +176,16 @@ private Config buildConfig() {
         return builder.build();
     }
 
+    private ConfigSource addConfigOrdinal(ConfigSource config, String ordinal) {
+        return addConfigOrdinal(config, "Map", ordinal);
+    }
+
+    private ConfigSource addConfigOrdinal(ConfigSource config, String type, String ordinal) {
+        Map<String, String> properties = new HashMap<>(config.getProperties());
+        properties.putIfAbsent(ConfigSource.CONFIG_ORDINAL, ordinal);
+        return MpConfigSources.create(type, properties);
```

Comment:
`@AddConfigSource` can be used to implement dynamic config.
This code is creating snapshot of the config source, which goes against the dynamic aspect.

Instead, you use a delegate and `getOrdinal()`:
- If `delegate.getOrdinal()` is the default method implementation
- And if `delegate.getOrdinal() == DEFAULT_ORDINAL` 
- then return the desired hard-coded ordinal